### PR TITLE
Constraints

### PR DESCRIPTION
**Pull request by tomjelinek**
add support for configuring resource constraints

### CODE REVIEW

File: tasks/create-and-push-cib.yml [hunk 1]
```diff
@@ -75,34 +75,114 @@
...
+        loop_var: constraint
 
-- name: Configure cluster resource clones
-  include_tasks: pcs-cib-resource-clone-pcs-0.10.yml
   vars:
-    resource_clone: "{{ item }}"
-  loop: "{{ ha_cluster_resource_clones }}"
-
+    __ha_cluster_pcs_cli_role:
+      promoted: "{%
```
Comment (richm) [inline tasks/create-and-push-cib.yml:174]: simpler to use `ternary`:
```
promoted: "{{ 'pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities |
                     ternary('Promoted', 'Master') }}"
unpromoted: "{{ 'pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities |
                     ternary('Unpromoted', 'Slave') }}"
```

File: tasks/pcs-cib-constraint-location.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        {% endif %}
+        {% for option in constraint.options | default([]) %}
+          {{ option.name | quote }}={{ option.value | quote }}
+        {% endfor %}
+        {{ constraint.rule }}
+      {% elif
+        (constraint.id | default())
+        or (
+          constraint.options | default([])
+          | rejectattr('name', 'eq', 'score') | list
```
Comment (richm) [inline tasks/pcs-cib-constraint-location.yml:35]: Can't use `eq` with jinja 2.7 - use this instead: `rejectaddr('name', 'match', '^score$')`

File: tests/tests_cib_constraints_create.yml [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+
+        - name: Fetch cluster versions of cluster components
+          include_tasks: tasks/fetch_versions.yml
+
+        # yamllint disable rule:line-length
+        - block:
+            - name: Fetch location constraints configuration from the cluster
+              command:
+                cmd: pcs constraint location --full
+              register: __test_pcs_location_config
```
Comment (richm) [inline tests/tests_cib_constraints_create.yml:402]: `changed_when: false`

File: tasks/create-and-push-cib.yml (hunk 1 above)
Comment (richm) [reply tasks/create-and-push-cib.yml:174]: Nevermind - I see you use `ternary` below - so if you want to keep the code as-is for readability purposes, that's fine with me

File: tests/tests_cib_constraints_create.yml (hunk 3 above)
Comment (richm) [reply tests/tests_cib_constraints_create.yml:402]: might need to add this to several other `command` uses in tests

File: tasks/create-and-push-cib.yml (hunk 1 above)
Comment (tomjelinek) [reply tasks/create-and-push-cib.yml]: Ternary actually looks better to me, so I went with that. Thanks for pointing it out.

File: tests/tests_cib_constraints_create.yml (hunk 3 above)
Comment (tomjelinek) [reply tests/tests_cib_constraints_create.yml:402]: I added `changed_when: no` to (hopefully) all 'fetch configuration' commands in tests.

File: tasks/pcs-cib-constraint-location.yml (hunk 2 above)
Comment (tomjelinek) [reply tasks/pcs-cib-constraint-location.yml]: Fixed, thanks for the hint!

**Comment by richm** [comment]
lgtm - and the tox tests are passing - let's wait for the integration test results - should be scheduled soon

**Comment by spetrosi** [comment]
[citest]

**Pull request merged**